### PR TITLE
Added linter to ensure HTTP requests are not made directly using requests or httpx

### PR DESCRIPTION
**Pull request by daladim**
This ensures we'll not forget to use the "right" (very opiniated) HTTP client API.

This is a follow-up of #132.

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -6,3 +6,11 @@ ignore=
     W503,
     # line break after binary operator
     W504,
+per-file-ignores =
+    # Our http module wraps httpx, and is the only place where calls to httpx are allowed
+    src/kraken/std/http/__init__.py:KRA001,
```
Comment (ntrinquier) [inline .flake8:11]: nice!

**Review by ntrinquier — APPROVED** [review]

File: src/tests/data/http_requests.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# This file has examples of constructs that should be disallowed by some of our custom linters
+
+import httpx
+import requests
+
+from kraken.std import http
+
+
+def please_dont_do_this_dear() -> None:
```
Comment (cdbrkfxrpt) [inline src/tests/data/http_requests.py:9]: now don't get me wrong - i love these :D

still, let's try to keep the test names descriptive, please

File: src/kraken/std/http/lint_ban_bare_requests.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+See https://github.com/kraken-build/kraken-std/pull/132
+"""
+
+
+class BareCallsVisitor(ast.NodeVisitor):
+    def __init__(self, disallowed_module: str) -> None:
+        self.bare_calls: list[ast.AST] = []
+        self.disallowed_module = disallowed_module
+
+    def visit_Call(self, node: ast.Call) -> None:
```
Comment (cdbrkfxrpt) [inline src/kraken/std/http/lint_ban_bare_requests.py:34]: typo in function signature, ought to be `visit_call` (non-capital c)

File: src/kraken/std/http/lint_ban_bare_requests.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        if BareCallsVisitor.is_bare_call(node, self.disallowed_module):
+            self.bare_calls.append(node)
+
+    @classmethod
+    def is_bare_call(cls, call: ast.Call, disallowed_module: str) -> bool:
```
Comment (cdbrkfxrpt) [inline src/kraken/std/http/lint_ban_bare_requests.py:39]: in this function name and that of `is_from_httpx_or_requests`, the thing you're referring to is the call but these functions are classmethods on `BareCallsVisitor`, which looks weird to me. i think i would call them `call_is_warbl_garbl` instead.

i'm nitpicking though :D

File: src/tests/data/http_requests.py (hunk 2 above)
Comment (daladim) [reply src/tests/data/http_requests.py:9]: These are not test names, this is a file that's not supposed to be run (but merely parsed by the linter). Still, I have added docstrings to be more explicit.

File: src/kraken/std/http/lint_ban_bare_requests.py (hunk 3 above)
Comment (daladim) [reply src/kraken/std/http/lint_ban_bare_requests.py:34]: I have just double checked. But this is expected, that's [how ast.NodeVisitor works](https://docs.python.org/3/library/ast.html#ast.NodeVisitor.visit) (its functions are `visit_classname`, and `Call` is a class from `ast`.

A lowercase `c` makes the linter fail

File: src/kraken/std/http/lint_ban_bare_requests.py (hunk 4 above)
Comment (daladim) [reply src/kraken/std/http/lint_ban_bare_requests.py]: I've changed the function names, thanks

File: src/kraken/std/http/lint_ban_bare_requests.py (hunk 3 above)
Comment (cdbrkfxrpt) [reply src/kraken/std/http/lint_ban_bare_requests.py:34]: ah lol

File: src/tests/data/http_requests.py (hunk 2 above)
Comment (cdbrkfxrpt) [reply src/tests/data/http_requests.py:9]: i'm still not a huge fan of these "cute" names in the codebase, but the docstrings serve their purpose so i'm resolving this ;)

**Pull request merged**